client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/RequestBuilder.java

```diff
@@ -16,10 +19,13 @@ public RequestBuilder(MethodInfo method) {
 
     String build() {
         request.append(method.getName());
-        if (method.getParameterCount() > 0) {
+        List<ParameterInfo> parameters = method.parameters()
+                .filter(parameterInfo -> !parameterInfo.isAnnotated(Header.class))
+                .collect(toList());
+        if (parameters.size() > 0) {
             request.append("(");
             Repeated repeated = new Repeated(", ");
-            for (ParameterInfo parameterInfo : method.getParameters()) {
+            for (ParameterInfo parameterInfo : parameters) {
```

Comment:
Maybe we could replace the `Repeated` code with `StringJoiner` or `Collectors.joining()` as suggested here: https://stackoverflow.com/a/669165/6575578

Reply:
I had that first, but it made the code more complex: appending to the StringBuilder in the different methods is easier to grasp.